### ISSUE TITLE
`UP029` triggers if import redefines previous import

### ISSUE THREAD

**Issue by injust** [astral-sh/ruff#16182]
### Description

```python
from math import *  # noqa: F403
from builtins import pow

pow(1, 1, 1)
```

https://play.ruff.rs/1f69e684-9b27-4626-ac06-4358cb6f9679

```python
from math import pow
from builtins import pow

pow(1, 1, 1)
```

https://play.ruff.rs/d86b74ac-4412-4ca6-a8ec-902a31dce30f

`from builtins import pow` should not trigger `UP029`, since the import is not a no-op in either case.

**Comment by ntBre** [comment]
Thanks for the report! I think the first case could be a bit tricky, especially if the `*` import is from outside the standard library, but the second case would be good to handle if nothing else.
